pkg/chunked/storage_linux.go

```diff
@@ -1530,6 +1546,11 @@ func (c *chunkedDiffer) ApplyDiff(dest string, options *archive.TarOptions, diff
 		c.fileType = fileTypeZstdChunked
 		c.manifest = manifest
 		c.tarSplit = tarSplit
+
+		if compressedDigest != *c.blobDigest {
```

Comment:
Check this before doing _anything_ with the file. At the very least immediately after `converTarToZstdChunked`, but … (A)